src/game.cpp

```diff
@@ -1629,10 +1629,25 @@ void Game::run()
 			&& client->checkPrivilege("fast");
 #endif
 
+	irr::core::dimension2d<u32> previous_screen_size(g_settings->getU16("screenW"),
+		g_settings->getU16("screenH"));
+
 	while (device->run()
 			&& !(*kill || g_gamecallback->shutdown_requested
 			|| (server && server->getShutdownRequested()))) {
 
+		// Verify if window size has changed and save it if it's the case
+		// Ensure evaluating settings->getBool after verifying screensize
+		// First condition is cheaper
+		if (previous_screen_size != device->getVideoDriver()->getScreenSize() &&
+			g_settings->getBool("autosave_screensize")) {
+			const irr::core::dimension2d<u32> &current_screen_size =
+				device->getVideoDriver()->getScreenSize();
```

Comment:
Would it hurt to move this variable definition outside the `if`, since the screen size is checked anyway?